.github/workflows/reusable-msi-build.yml

```diff
@@ -12,23 +12,23 @@ env:
 
 jobs:
   msi-build:
-    runs-on: ubuntu-24.04
+    runs-on: ${{ inputs.OS }}
```

Comment:
In principle this could be a single OS but since the callers of this reusable workflow are already using different OS versions I used it as a way to run on different hosts. Anyway, this seems something that we could potentially remove, but, that should be a separate PR since it touches other workflow fils.